sdk/cosmos/src/clients/document_client.rs

```diff
@@ -85,9 +81,45 @@ impl DocumentClient {
         GetDocumentResponse::try_from(response).await
     }
 
+    /// replace a document in a collection
+    pub async fn replace_document<'a, T: Serialize>(
+        &self,
+        ctx: Context,
+        document: &'a T,
+        options: ReplaceDocumentOptions<'_>,
```

Comment:
It's not something to address in this PR, but I wonder if instead of accepting a `xxxOptions` we should accept an `Into<xxxOptions>`. 
This way we could implement, for example, `Into<ReplaceDocumentOptions> for ConsistencyLevel` and  make the code less noisy.